plugins/com.google.cloud.tools.eclipse.appengine.libraries/src/com/google/cloud/tools/eclipse/appengine/libraries/model/MavenCoordinates.java

```diff
@@ -126,35 +120,28 @@ public Builder toBuilder() {
     private String version = LATEST_VERSION;
     private String type = JAR_TYPE;
     private String classifier;
-
-    public MavenCoordinates build() {
-      MavenCoordinates coordinates = new MavenCoordinates(groupId, artifactId); 
-      coordinates.repository = repository;
-      coordinates.version = version;
-      coordinates.type = type;
-      coordinates.classifier = classifier;
-      return coordinates;
-    }
     
     /**
      * @param groupId the Maven group ID
+     * @param artifactId the Maven artifact ID
      */
-    public Builder setGroupId(String groupId) {
+    public Builder(String groupId, String artifactId) {
```

Comment:
I wonder if the constructor should be empty with named setters for groupId and artifactId? One of the advantages of the builder pattern is named parameters, which this loses.

Reply:
I see. So the pattern does want to have a 0-arg constructor. I'll revert the changes.